Library/Homebrew/formula_installer.rb

```diff
@@ -1379,18 +1379,31 @@ def forbidden_license_check
     EOS
   end
 
+  sig { params(tap: Tap, allowed_taps: T::Set[Tap], forbidden_taps: T::Set[Tap]).returns(T::Boolean) }
+  def allowed_tap?(tap, allowed_taps, forbidden_taps)
+    (tap.official? || allowed_taps.blank? || allowed_taps.include?(tap)) && forbidden_taps.exclude?(tap)
+  end
```

Comment:
I should probably move this into the `Tap` class so we can just do `Tap#allowed?`.

Reply:
@carlocab Yeh, I'd be happy with that either way. May want to do something like `allowed_by_env?`